tests/test_requests.py

```diff
@@ -221,6 +221,41 @@ def test_http_303_doesnt_change_head_to_get(self, httpbin):
         assert r.history[0].status_code == 303
         assert r.history[0].is_redirect
 
+    def test_header_and_body_removal_on_redirect(self, httpbin):
+        purged_headers = ('Content-Length', 'Content-Type')
+        ses = requests.Session()
+        req = requests.Request('POST', httpbin('post'), data={'test': 'data'})
+        prep = ses.prepare_request(req)
+        resp = ses.send(prep)
+
+        # Mimic a redirect response
+        resp.status_code = 302
+        resp.headers['location'] = 'get'
+
+        # Run request through resolve_redirects
+        next_resp = next(ses.resolve_redirects(resp, prep))
+        assert next_resp.request.body is None
+        for header in purged_headers:
+            assert header not in next_resp.request.headers
+
+    def test_transfer_enc_removal_on_redirect(self, httpbin):
+        purged_headers = ('Transfer-Encoding', 'Content-Type')
+        ses = requests.Session()
+        req = requests.Request('POST', httpbin('post'), data=(b'x' for x in range(1)))
+        prep = ses.prepare_request(req)
+        assert 'Transfer-Encoding' in prep.headers
+        resp = ses.send(prep)
```

Comment:
I reduced the `range` call to size 1 because we were hitting [this bug](https://github.com/kevin1024/pytest-httpbin/issues/33) when sending more than one chunk. After some more testing, this only passes *most of the time*, which likely isn't something we want to introduce. 

The failures seem random and are tied specifically to the httpbin module. Calling the live http://httpbin.org/post endpoint won't produce the failure.

As I see it, two options are either calling the httpbin.org/post endpoint directly (probably not the best solution) or adding a try/except around `resp = ses.send(prep)` and manually creating the Response on failure.

Thoughts?